### PR TITLE
Let `CMD <cmd>` in the Dockerfile template be configurable

### PR DESCRIPTION
**Pull request by AliMirlou**
A configurable CMD might not be something this project wants but there are many use-cases for this, like starting the notebook in a different way or even using Repo2Docker as a library to build non-jupyter Docker images. It would be nice to have this feature.

By the way, this change is completely **backward-compatible** and **does not affect** anyone who does not want to change the CMD.

### CODE REVIEW

**Comment by AliMirlou** [comment]
I haven't reflected the changes in the documentations and changelog yet. Waiting for someone's approval first.

**Comment by AliMirlou** [comment]
Also can anyone explain why the "lint" check is failing? Is it because of too many characters in one line?

**Comment by manics** [comment]
We use pre-commit and the black autoformatter.

If you install pre-commit (e.g. with pip) and run

    pre-commit run --all-files

the code should pass the CI lint check.

**Comment by AliMirlou** [comment]
@manics Can you review this PR yourself? It's not a major change.

**Comment by AliMirlou** [comment]
@betatim I need some approval to continue this PR. Can you give a quick review?

**Comment by AliMirlou** [comment]
Or @minrk perhaps?

**Comment by AliMirlou** [comment]
@manics Can you review this PR now?

File: repo2docker/buildpacks/base.py
Comment (consideRatio) [inline repo2docker/buildpacks/base.py:489]:
```suggestion
    def get_cmd(self):
        """
        A list of strings to be used for the Dockerfile's `CMD`.
```

File: repo2docker/buildpacks/base.py
Comment (consideRatio) [inline repo2docker/buildpacks/base.py:576]: - Removed validation as this isn't user facing, just for those making custom buildpacks. My goal is to minimize complexity and ease the long term maintenance and code comprehension by doing that.
- Removed tuples fixup for similar reasons.

```suggestion
        cmd = self.get_command()
        if isinstance(cmd, list):
            cmd = str(cmd).replace("'", '"')
```

I don't understand what is going on with regards to the replacement operation that remains, but I suspect its not reasonable to provide either.

File: repo2docker/buildpacks/base.py
Comment (consideRatio) [inline repo2docker/buildpacks/base.py:592]:
```suggestion
            cmd=cmd,
```

File: repo2docker/buildpacks/base.py
Comment (consideRatio) [inline repo2docker/buildpacks/base.py:194]:
```suggestion
CMD {{ cmd }}
```

**Review by consideRatio — CHANGES_REQUESTED** [review]
I think it could be acceptable to provide this, but I'm not sure.

I made a few suggestions towards making me feel comfortable with the implementation. I think a very strong case needs to be made for a string replacement still though, and it probably needs to be documented if a strong case for it is presented.

**Comment by consideRatio** [comment]
## Related
- https://github.com/jupyterhub/repo2docker/issues/1206

**Comment by manics** [comment]
Thanks for the PR. I'm closing since there's been no updates, and it's already possible to override `CMD` using appendix
https://repo2docker.readthedocs.io/en/latest/usage.html#cmdoption-jupyter-repo2docker-appendix

**Pull request closed without merging**
